api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/ReLUTest.kt

```diff
@@ -10,36 +10,27 @@ import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Test
 
-private const val DATA_SIZE = 4
-
 internal class ReLUTest : ActivationLayerTest() {
     @Test
     fun defaultRelu() {
         val input = floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f)
         val expected = floatArrayOf(0.0f, 0.0f, 0.0f, 2.0f)
-        val actual = floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f)
 
-        assertActivationFunction(ReLU(), input, actual, expected)
+        assertActivationFunctionSameOutputShape(ReLU(), input, expected)
     }
 
     @Test
     fun reluWithMaxValue() {
         val input = floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f)
         val expected = floatArrayOf(0.0f, 0.0f, 0.0f, 1.0f)
-        val actual = floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f)
 
-        assertActivationFunction(ReLU(maxValue = 1.0f), input, actual, expected)
+        assertActivationFunctionSameOutputShape(ReLU(maxValue = 1.0f), input, expected)
     }
 
     @Test
     fun reluWithNegativeSlope() {
-        val input = floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f)
-        val expected = floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f)
-        val actual = floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f)
-
-
         val exception = Assertions.assertThrows(UnsupportedOperationException::class.java) {
-            assertActivationFunction(ReLU(negativeSlope = 1.0f), input, actual, expected)
+            assertActivationFunctionIrrelevantInputOutput(ReLU(negativeSlope = 1.0f))
```

Comment:
Remark: I think it's a good idea for tests, we need more special testing functions to reuse test-scenarios

Reply:
will try to introduce some if see any possibility to abstract here 👍🏻